src/ynEIGEN/EigenStrategyManager.sol

```diff
@@ -0,0 +1,352 @@
+        tokenStakingNodesManager = init.tokenStakingNodesManager;
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  STRATEGY  ----------------------------------------
+    //--------------------------------------------------------------------------------------
+    
+    /**
+     * @notice Stakes specified amounts of assets into a specific node on EigenLayer.
+     * @param nodeId The ID of the node where assets will be staked.
+     * @param assets An array of ERC20 tokens to be staked.
+     * @param amounts An array of amounts corresponding to each asset to be staked.
+     */
+    function stakeAssetsToNode(
+        uint256 nodeId,
+        IERC20[] calldata assets,
+        uint256[] calldata amounts
+    ) external onlyRole(STRATEGY_CONTROLLER_ROLE) nonReentrant {
+        uint256 assetsLength = assets.length;
+        uint256 amountsLength = amounts.length;
+
+        if (assetsLength != amountsLength) {
+            revert LengthMismatch(assetsLength, amountsLength);
+        }
+
+        ITokenStakingNode node = tokenStakingNodesManager.getNodeById(nodeId);
+        if (address(node) == address(0)) {
+            revert InvalidNodeId(nodeId);
+        }
+
+        IStrategy[] memory strategiesForNode = new IStrategy[](assetsLength);
+        for (uint256 i = 0; i < assetsLength; i++) {
+            IERC20 asset = assets[i];
+            if (amounts[i] == 0) {
+                revert InvalidStakingAmount(amounts[i]);
+            }
+            IStrategy strategy = strategies[asset];
+            if (address(strategy) == address(0)) {
+                revert StrategyNotFound(address(asset));
+            }
+            strategiesForNode[i] = strategy;
+        }
+        // Transfer assets to node
+        ynEigen.retrieveAssets(assets, amounts);
+
+        IERC20[] memory depositAssets = new IERC20[](assetsLength);
+        uint256[] memory depositAmounts = new uint256[](amountsLength);
+
+        for (uint256 i = 0; i < assetsLength; i++) {
+            (IERC20 depositAsset, uint256 depositAmount) = toEigenLayerDeposit(assets[i], amounts[i]);
+            depositAssets[i] = depositAsset;
+            depositAmounts[i] = depositAmount;
+
+            // Transfer each asset to the node
+            depositAsset.transfer(address(node), depositAmount);
+        }
+
+        emit StakedAssetsToNode(nodeId, assets, amounts);
+
+        node.depositAssetsToEigenlayer(depositAssets, depositAmounts, strategiesForNode);
+
+        emit DepositedToEigenlayer(depositAssets, depositAmounts, strategiesForNode);
+    }
+
+    function toEigenLayerDeposit(
+        IERC20 asset,
+        uint256 amount
+    ) internal returns (IERC20 depositAsset, uint256 depositAmount) {
+        if (address(asset) == address(wstETH)) {
+            // Adjust for wstETH
+            depositAsset = IERC20(wstETH.stETH());
```

Comment:
The stETH and oETH address is unlikely to change and can be set as a constant.

Reply:
we passed them in as params in the end to make our holesky deployment easy.

We Maintain a canonical version there.